src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -689,3 +689,20 @@ def extensions(self):
             extensions.append(x509.Extension(oid, critical, value))
 
         return x509.Extensions(extensions)
+
+    def public_bytes(self, encoding):
+        if not isinstance(encoding, serialization.Encoding):
+            raise TypeError("encoding must be an item from the Encoding enum")
+
+        # TODO: make text prelude optional.
+        bio = self._backend._create_mem_bio()
+        if encoding is serialization.Encoding.PEM:
+            res = self._backend._lib.X509_REQ_print(bio, self._x509_req)
```

Comment:
I'd like to keep the text form somehow though as it's super useful for debugging (it has already proved useful in the test vector PRs).  Do you think I should put the `_print` version in a `__str__` implementation or something?